src/geode/geometry/intersection.cpp

```diff
@@ -838,14 +839,172 @@ namespace geode
+                                             &circle](
+                                             const Point3D& intersection ) {
+            auto plane_output = point_plane_distance( intersection, plane );
+            correctness.first.first =
+                std::get< 0 >( plane_output ) <= global_epsilon;
+            correctness.first.second.emplace_back(
+                std::move( std::get< 1 >( plane_output ) ) );
+            auto circle_output = point_circle_distance( intersection, circle );
+            correctness.second.first =
+                std::get< 0 >( circle_output ) <= global_epsilon;
+            correctness.second.second.emplace_back(
+                std::move( std::get< 1 >( circle_output ) ) );
+        };
+        if( discr == 0. )
+        {
+            // The quadratic polynomial has 1 real-valued repeated root.
+            // The circle just touches the plane.
+            compute_correctness( result.emplace_back(
+                line.origin() - line.direction() * ( a1 / a2 ) ) );
+        }
+        else
+        {
+            // The quadratic polynomial has 2 distinct, real-valued roots.
+            // The circle intersects the plane in two points.
+            const auto root = std::sqrt( discr );
+            compute_correctness( result.emplace_back(
+                line.origin() - line.direction() * ( ( a1 + root ) / a2 ) ) );
+            compute_correctness( result.emplace_back(
+                line.origin() - line.direction() * ( ( a1 - root ) / a2 ) ) );
+        }
+        return { std::move( result ), std::move( correctness ) };
+    }
+
+    IntersectionResult< InfiniteLine3D > plane_plane_intersection(
+        const Plane& plane0, const Plane& plane1 )
+    {
+        // If N0 and N1 are parallel, either the planes are parallel and
+        // separated or the same plane.
+        //  Otherwise, the intersection line is
+        //   L(t) = t*Cross(N0,N1)/|Cross(N0,N1)| + c0*N0 + c1*N1
+        // for some coefficients c0 and c1 and for t any real number (the
+        // line parameter).  Taking dot products with the normals,
+        //   d0 = Dot(N0,L) = c0*Dot(N0,N0) + c1*Dot(N0,N1) = c0 + c1*d
+        //   d1 = Dot(N1,L) = c0*Dot(N0,N1) + c1*Dot(N1,N1) = c0*d + c1
+        // where d = Dot(N0,N1).  These are two equations in two unknowns.
+        // The solution is
+        //   c0 = (d0 - d*d1)/det
+        //   c1 = (d1 - d*d0)/det
+        // where det = 1 - d^2.
+        const auto dot = plane0.normal().dot( plane1.normal() );
+        const auto constant0 = plane0.plane_constant();
+        const auto constant1 = plane1.plane_constant();
+        if( std::fabs( dot ) >= 1. )
+        {
+            // The planes are parallel.  Check if they are coplanar.
+            const auto constant_diff =
+                dot >= 0. ? constant0 - constant1 : constant0 + constant1;
+            if( std::fabs( constant_diff ) == 0. )
```

Comment:
```suggestion
            if( constant_diff == 0. )
```
could it fail if it returns -0?

Reply:
I do not know, took this code from geometricTools